Objects/typeobject.c

```diff
@@ -3289,7 +3290,11 @@ type_setattro(PyTypeObject *type, PyObject *name, PyObject *value)
     }
     res = _PyObject_GenericSetAttrWithDict((PyObject *)type, name, value, NULL);
     if (res == 0) {
-        res = update_slot(type, name);
+        if (name_looks_like_slot(name)) {
+            res = update_slot(type, name);
+        } else {
+            PyType_Modified(type);
```

Comment:
Maybe move `PyType_Modified()` out of `update_slot()`? Then it could be called unconditionally here and only once in `update_all_slots()`.

Reply:
I think that's a good idea.